test/autorest/generated/lrogroup/lroretrys.go

```diff
@@ -184,7 +244,25 @@ func (client *lroRetrysOperations) post202Retry200HandleResponse(resp *azcore.Re
 
 // PostAsyncRelativeRetrySucceeded - Long running post request, service returns a 500, then a 202 to the initial request, with an entity that contains ProvisioningState=’Creating’. Poll the endpoint indicated in the Azure-AsyncOperation header for operation status
 func (client *lroRetrysOperations) BeginPostAsyncRelativeRetrySucceeded(ctx context.Context, lroRetrysPostAsyncRelativeRetrySucceededOptions *LroRetrysPostAsyncRelativeRetrySucceededOptions) (LroRetrysPostAsyncRelativeRetrySucceededPoller, error) {
+	req, err := client.postAsyncRelativeRetrySucceededCreateRequest(lroRetrysPostAsyncRelativeRetrySucceededOptions)
+	if err != nil {
+		return nil, err
+	}
+	// send the first request to initialize the poller
+	resp, err := client.p.Do(ctx, req)
+	if err != nil {
+		return nil, err
+	}
+	pt := pollingTrackerPost{
```

Comment:
I would move this creation logic into an internal method in pollers_helpers.go that creates the tracker and calls `initializeState()` (you can get the request verb from the resp object).

Reply:
Looks like this method already exists createPollingTracker, just need to invoke it?

Reply:
So I made this change, but it involved declaring the pollingTracker interface instead of the concrete poller type (pollingTrackerDelete for instance) in each of the internal poller structs that we are creating for each operation. I think it's fine to use the pollingTracker interface which is why I made the change. If you want to revert or discuss let me know.

Reply:
Yep I think that's great.